VRDR.Tests/Filtering_Should.cs

```diff
@@ -0,0 +1,341 @@
+            {
+                string postFilteredBaseMessage = noFieldsFilterService.filterMessage(preFilteredBaseMessage).ToJson();
+                BaseMessage.Parse(postFilteredBaseMessage);
+            }
+            catch (Exception e)
+            {
+                throw new Exception(e.Message);
+            }
+        }
+        
+        [Fact]
+        public void FilterAllFields_3()
+        {
+            BaseMessage preFilteredBaseMessage = BaseMessage.Parse(File.ReadAllText("./fixtures/filter/LIMITS-test_1.json"));
+
+            try
+            {
+                string postFilteredBaseMessage = noFieldsFilterService.filterMessage(preFilteredBaseMessage).ToJson();
+                BaseMessage.Parse(postFilteredBaseMessage);
+            }
+            catch (Exception e)
+            {
+                throw new Exception(e.Message);
+            }
+        }
+
+        [Fact]
+        public void FilterFilePerJurisdictionFilters()
+        {
+            List<string> passedJurisdictions = new List<string>();
+            List<string> failedJurisdictions = new List<string>();
+
+            BaseMessage preFilteredBaseMessage = BaseMessage.Parse(File.ReadAllText("./fixtures/filter/Pre-filtered-file.json"));
+
+            string jurisdictionFilterText = File.ReadAllText("./fixtures/filter/JurisdictionFieldsIJEFilter.json");
+            Dictionary<string, string[]> jurisdictionFilterJSON = JsonConvert.DeserializeObject<Dictionary<string, string[]>>(jurisdictionFilterText);
+
+            string[] jurisdictionArray = jurisdictionFilterJSON.Keys.ToArray();
+            foreach (var jurisdictionId in jurisdictionArray)
+            {
+                string jurisdictionFilter = JsonConvert.SerializeObject(jurisdictionFilterJSON[jurisdictionId]);
+                
+                FilterService jurisdictionFilterService = new FilterService(jurisdictionFilter, "./fixtures/filter/IJEToFHIRMapping.json", false);
+                
+                try
+                {
+                    string postFilteredBaseMessage = jurisdictionFilterService.filterMessage(preFilteredBaseMessage).ToJson();
+                    BaseMessage.Parse(postFilteredBaseMessage);
+                    passedJurisdictions.Add(jurisdictionId);
+                }
+                catch (Exception e)
+                {
+                    failedJurisdictions.Add(jurisdictionId);
+                    throw new Exception(e.Message);
+                }
+            }
+            
+            Console.WriteLine("=============Jurisdiction filtering test=============");
+            Console.WriteLine($"Passed jurisdiction count = {passedJurisdictions.Count} ({JsonConvert.SerializeObject(passedJurisdictions)})");
+            Console.WriteLine($"Failed jurisdiction count = {failedJurisdictions.Count} ({JsonConvert.SerializeObject(failedJurisdictions)})");
+        }
+        
+        private static DeathRecord getDeathRecordFromMessage(BaseMessage message)
```

Comment:
Parsing already extracts the record, as in VRDR.CLI/Program.cs "extract" option code below.  Suggest adding DeathRecordUpdateMessage to this code below, refactoring it to the BaseMessage class, then remove "getDeathRecordFromMessage()" here and use the refactored BaseMessage method.

            else if (args.Length == 2 && args[0] == "extract")
            {
                BaseMessage message = BaseMessage.Parse(File.ReadAllText(args[1]));
                DeathRecord record;
                switch (message)
                {
                    case DeathRecordSubmissionMessage submission:
                        record = submission.DeathRecord;
                        Console.WriteLine(record.ToJSON());
                        break;
                    case CauseOfDeathCodingMessage coding:
                        record = coding.DeathRecord;
                        Console.WriteLine(record.ToJSON());
                        break;
                    case DemographicsCodingMessage coding:
                        record = coding.DeathRecord;
                        Console.WriteLine(record.ToJSON());
                        break;
                }
                return 0;
            }

Reply:
I just want to confirm what you mean by this comment.
Add the case statement for DeathRecordUpdateMessage.
Remove the method, getDeathRecordFromMessage() completely.
I am uncertain about refactoring to the BaseMessage class and how to use this logic from the test class.

Reply:
Did you see my comment here?